### ISSUE TITLE
Completely Functional... But No Alerts.

### ISSUE THREAD

**Issue by OlivePip** [WildcardSearch/MentionMe#208]
Hi, I've had this issue for months now and nothing seems to fix it. I've reinstalled the plugin multiple times, I've even done a clean install of Mybb with only the pluginlibrary and MyAlerts plugins installed but the issue persists. Besides changing a few board settings, the only customized aspect of the freshly installed board is the theme but the issue carries over to the default Mybb skin anyway. All plugins are up to date.

The option to receive alerts for mentions has been selected at all times, although I have tried deselecting and reselecting to make it work. The plugin says it has successfully integrated. Other alerts work perfectly and this issue was the same on the original install when I had the DVZ mentions plugin installed initially.

I'm usually quite good at rooting out the source of an issue like this without having to seek help but I'm drawing a serious blank and I can't seem to find anything by googling. It's probably something really obvious for which I'm sorry if I've wasted your time, but I'd be eternally grateful if you could possibly help!

**Comment by WildcardSearch** [comment]
I'm sorry I am so slow to reply, but I am working a lot of hours right now.

When things return to normal (idk when that will be), I will be glad to help you.

Sorry you are having problems.

**Comment by OlivePip** [comment]
Oh great, thanks! I thought I'd gone and stuck this in the wrong place so I sheepishly closed it and reposted the issue on Mybb. [Here's the link to the topic.](https://community.mybb.com/thread-215691.html)

**Comment by WildcardSearch** [comment]
Because you mentioned that DVZ Mentions also has the same problem, it makes me think that the problem has to do with your forum, another plugin, or something of that nature.

Monday I will go back to a more reasonable schedule at work. When that happens, I will try to give you some help, which may require you to provide me with access to your board.

Again, I am sorry that you are having issues.

**Comment by OlivePip** [comment]
I also thought it was likely to be a problem with the board or another plugin due to both mention plugins not functioning, but freshly installing everything and only installing the relevant plugins didn't help which is what really baffled me.

Fortunately the site is in beta mode at the moment so it's not causing too much of an inconvenience yet!

**Comment by WildcardSearch** [comment]
Until I can get a chance to look at it, you could try looking through your error logs to see if there are any issues. Are you familiar with how to do that?

**Comment by OlivePip** [comment]
If it's the error log at /error.log then there isn't one at the moment - I set the board to make error logs just now since that wasn't selected prior. There's currently nothing although I have tried to use the mentions since to try and prompt an error log to be made. Is that the error log you meant?

**Comment by WildcardSearch** [comment]
Yes. That at least helps to eliminate errors in missing files and such.

I'm currently battling the flu (as if I needed more problems right now). I'll try to contact you tm to see if we can figure it out, but forgive me if I am unable to do so.

**Comment by OlivePip** [comment]
I hope you're feeling better! 

So I think the problem may be with MyAlerts itself as I've just discovered that PMs are not consistently producing alerts either (although PMs on a attached accounts alert seems to be) but it could be a coincidence.

**Comment by WildcardSearch** [comment]
Thanks for the update. Unfortunately, I am still down with the flu. I thought it would be over by now...

**Comment by OlivePip** [comment]
I hope you're feeling better! I still haven't had any luck figuring out the problem and if it is MyAlerts at the root of the problem which I suspect it is, I'm not sure what to do about it as the support/updates for it seem to be on hold. The error log has finally produced something, but I don't think it's relevant or helpful sadly: 
`<error>
	<dateline>1517937069</dateline>
	<script>inc/datahandlers/user.php</script>
	<line>191</line>
	<type>2</type>
	<friendly_type>Warning</friendly_type>
	<message>strpos(): Empty needle</message>
</error>`

**Comment by WildcardSearch** [comment]
No, I can't reproduce it at all.

I'm running the latest version of both MentionMe and MyAlerts on Rant Central and I get alerts from everything...

**Comment by Eldenroot** [comment]
OK, thank you for your tests. OK, again something wrong on my board :( I will try to debug this ;)

**Comment by helloshago** [comment]
Hi Mark!

I'm a forum administrator over at www.charmingrp.com. For the past year, we've been struggling to get mentionme to work with myalerts. My co-admin is OlivePip, the starter of the thread.

We're still having the exact same issue and are really looking to get it fixed! Do you have time in your schedule to work on this for us? We are 100% willing to offer compensation for your time and appreciate your attention on this matter. I sent an email to the address you have listed in your bio.

**Comment by WildcardSearch** [comment]
@helloshago I tried to reply to your email but there was some issue. If you join the MyBB Discord, I will be listed near the top of any channel. Send me a DM and we can try to troubleshoot this issue together.

I'm sorry that you are having problems.

**Comment by helloshago** [comment]
Thank you for the quick response! My coadmin Olive is going to be messaging you on Discord as she’s been the one troubleshooting and is more tech savvy. What are you expecting as far as compensation so I can have it prepared?

**Comment by WildcardSearch** [comment]
ONE BILLION DOLLARS!!!

(I never charge 👅 )

**Comment by OlivePip** [comment]
Thanks again for all your help! Just in case anyone else has this exact issue, the problem stemmed from the MyAlerts plugin and there being no "Alert Types" option on the ACP Configuration menu. This was solved by editing the admin permissions to allow them to change alert types. Alert Types then appeared on the config menu and after reinstalling MentionMe, showed mentions as checked and problem solved!

I think this problem may have come about either through a bug in the initial MyAlerts installation or because of database changes I made when setting up the board to make certain user options set to default for new accounts.
